presto-resource-group-managers/src/main/java/com/facebook/presto/resourceGroups/ResourceEstimatesRange.java

```diff
@@ -0,0 +1,176 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.resourceGroups;
+
+import com.facebook.presto.spi.session.ResourceEstimates;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import io.airlift.units.DataSize;
+import io.airlift.units.Duration;
+
+import java.util.Objects;
+import java.util.Optional;
+
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+
+public final class ResourceEstimatesRange
+{
+    private final Optional<Range<Duration>> executionTimeRange;
+    private final Optional<Range<Duration>> cpuTimeRange;
+    private final Optional<Range<DataSize>> peakMemoryRange;
+
+    @JsonCreator
+    public ResourceEstimatesRange(
+            @JsonProperty("executionTimeRange") Optional<Range<Duration>> executionTimeRange,
+            @JsonProperty("cpuTimeRange") Optional<Range<Duration>> cpuTimeRange,
+            @JsonProperty("peakMemoryRange") Optional<Range<DataSize>> peakMemoryRange)
+    {
+        this.executionTimeRange = requireNonNull(executionTimeRange, "executionTimeRange is null");
+        this.cpuTimeRange = requireNonNull(cpuTimeRange, "cpuTimeRange is null");
+        this.peakMemoryRange = requireNonNull(peakMemoryRange, "peakMemoryRange is null");
+    }
+
+    @JsonProperty
+    public Optional<Range<Duration>> getExecutionTimeRange()
+    {
+        return executionTimeRange;
+    }
+
+    @JsonProperty
+    public Optional<Range<Duration>> getCpuTimeRange()
+    {
+        return cpuTimeRange;
+    }
+
+    @JsonProperty
+    public Optional<Range<DataSize>> getPeakMemoryRange()
+    {
+        return peakMemoryRange;
+    }
+
+    boolean match(ResourceEstimates resourceEstimates)
+    {
+        if (executionTimeRange.isPresent() &&
+                !(resourceEstimates.getExecutionTime().isPresent() && executionTimeRange.get().contains(resourceEstimates.getExecutionTime().get()))) {
+            return false;
+        }
+
+        if (cpuTimeRange.isPresent() &&
+                !(resourceEstimates.getCpuTime().isPresent() && cpuTimeRange.get().contains(resourceEstimates.getCpuTime().get()))) {
+            return false;
+        }
+
+        if (peakMemoryRange.isPresent() &&
+                !(resourceEstimates.getPeakMemory().isPresent() && peakMemoryRange.get().contains(resourceEstimates.getPeakMemory().get()))) {
+            return false;
+        }
+
+        return true;
+    }
+
+    @Override
+    public String toString()
+    {
+        final StringBuilder sb = new StringBuilder("ResourceEstimatesRange{");
```

Comment:
You can use `MoreObjects.toStringHelper()`. Also you don't need to mark this local variable as `final`.